src/types.ts

```diff
@@ -48,9 +48,28 @@ export interface Component {
   type: string;
 }
 
-export interface PrefabComponent {
-  actions?: PrefabAction[];
+export const isComponentTypePrefab = (
+  component: PrefabComponent,
+): component is ComponentTypePrefab =>
+  typeof component.type === 'undefined' || component.type === 'COMPONENT';
+export const isPartialTypePrefab = (
+  component: PrefabComponent,
+): component is PartialTypePrefab => component.type === 'PARTIAL';
+
+export type PrefabComponent = ComponentTypePrefab | PartialTypePrefab;
+
+export type ComponentTypePrefab = BasePrefab & {
+  type?: 'COMPONENT';
   descendants: PrefabComponent[];
+};
+
+export type PartialTypePrefab = BasePrefab & {
+  type: 'PARTIAL';
+  partialId: string;
```

Comment:
I think you can rename this to just `id`

Reply:
the partialId option has been changed to id